compiler/codegen/library.cpp

```diff
@@ -232,12 +238,39 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
           libname.c_str());
 
   //
-  // The ZMQ library has to be linked against the C++ stdlib. Rather than
-  // package libc++ up as part of our multi-locale library client when
-  // packaging it up, we'll let library-makefile do it for us.
+  // Multi-locale libraries require some extra libraries to be linked in order
+  // to function correctly. For static libraries in particular, rather than
+  // try to link these dependencies at compile time, we shunt responsibility
+  // off to the user via use of `--library-makefile`.
   //
   if (fMultiLocaleInterop) {
-    fprintf(makefile.fptr, " %s", "-lc++");
+
+    //
+    // Get the path to the launcher library.
+    //
+    std::string launcherLib = getCompilelineOption("launcher-libdir");
+    removeTrailingNewline(launcherLib);
+
+    fprintf(makefile.fptr, " -L%s", launcherLib.c_str());
+
+    fprintf(makefile.fptr, " %s", "-lchpllaunch");
+
+    //
+    // For right now, we assume GNU/Linux and use "libstdc++" unless we are
+    // on Macs, in which case we will try to link against "libc++" instead as
+    // it is the preferred default.
+    //
+    const char* cppStdLib = "-lstdc++";
+    if (!strcmp(CHPL_TARGET_PLATFORM, "darwin")) {
+      cppStdLib = "-lc++";
+    }
+
+    fprintf(makefile.fptr, " %s", cppStdLib);
+
+    //
+    // For right now, assume that ZMQ is on the system install path.
+    //
+    fprintf(makefile.fptr, " %s", "-lzmq");
```

Comment:
I might consider making all of this part of the new `compileline` flag - that way if the user feels like still using `compileline` instead of a Makefile, they can just make one additional call instead of this additional call + `-lstdc++ -lzmq`.

For instance, they would be able to say:
```bash
`$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries` `$CHPL_HOME/util/config/compileline --launcher-libdir`
```
instead of:
```bash
`$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries` `$CHPL_HOME/util/config/compileline --launcher-libdir` -lstdc++ -lzmq
```

Reply:
You would then likely want a different name for the `compileline` flag